substrate-node/pallets/pallet-smart-contract/src/cost.rs

```diff
@@ -239,13 +246,14 @@ pub fn calculate_discount<T: Config>(
         );
     }
 
-    let balance_as_u128: u128 = balance.saturated_into::<u128>();
-
     // calculate amount due on a monthly basis
-    // we bill every one hour so we can infer the amount due monthly (30 days ish)
-    let amount_due_monthly = amount_due * 24 * 30;
+    // fist get the normalized amount for one hour so we can infer the amount due monthly (30 days ish)
+    let amount_due_hourly = U64F64::from_num(amount_due) * U64F64::from_num(seconds_elapsed)
+        / U64F64::from_num(SECS_PER_HOUR);
```

Comment:
Maybe check the billing frequency instead?

Reply:
I get it
Problem is that the billing time can be different from billing frequency
See here:

https://github.com/threefoldtech/tfchain/blob/fc5e1820d554c9c48ee963dff53388d0514c9011/substrate-node/pallets/pallet-smart-contract/src/lib.rs#L1144

That why I made it more generic